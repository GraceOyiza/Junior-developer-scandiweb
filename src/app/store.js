import { configureStore } from '@reduxjs/toolkit';
import cartReducer from '../feature/cartSlice';
import currencyReducer from '../feature/currencySlice';
import productReducer from '../feature/productSlice';
import categoryReducer from '../feature/categorySlice';

export default configureStore({
  reducer: {
    cart: cartReducer,
    currency: currencyReducer,
    product: productReducer,
    category: categoryReducer,
  },
});