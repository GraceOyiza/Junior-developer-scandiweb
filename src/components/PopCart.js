import React from 'react';
import {  Popupcart,  ButtonClick, Button2, Overlay} from '../styled/PopCartStyled';
import {
  togglePopUpCart,
  incrementQuantity,
  decrementQuantity,
  setVariant,
} from '../feature/cartSlice';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import CartItem from './CartItem';
import { calculateCartTotal } from '../utils/common';





class PopCart extends React.Component {
  

  handleOverlayClick = () => {
    this.props.togglePopUpCart();
  };

  addToCart = () => {
    this.props.addToCart(this.props.cartItem);
  };

  selectVariant = (attribute, selectedValue) => {
    this.props.setVariant({ id: attribute, value: selectedValue });
  };

  closePopCart = () => this.props.togglePopUpCart();

  render() {
    return (
      <div>
        <Overlay isOpen={this.props.isOpen} onClick={this.handleOverlayClick} />
        <Popupcart isOpen={this.props.isOpen}>
          <div />
          <h3>My Bag {this.props.cartItems.length}</h3>
          <div>
            {this.props.cartItems.map((cartItem, index) => (
              <CartItem
                key={index}
                cartItem={cartItem}
                incrementQuantity={this.props.incrementQuantity}
                decrementQuantity={this.props.decrementQuantity}
                selectVariant={this.selectVariant}
              />
            ))}

            <div>
              <p>
                Total:{' '}
                {calculateCartTotal(
                  this.props.cartItems,
                  this.props.currentCurrency,
                )}
              </p>
              <Link onClick={this.closePopCart} to={`/cart`}>
                <Button2>VIEW BAG</Button2>
              </Link>

              <ButtonClick>CHECK OUT</ButtonClick>
            </div>
          </div>
        </Popupcart>
      </div>
    );
  }
}
 

// export default PopCart;

const mapStateToProps = state => ({
  cartItems: state.cart.cartItems,
  isOpen: state.cart.isOpen,
  currentCurrency: state.currency.currentCurrency || '$',
});
const mapDispatchToProps = {
  togglePopUpCart,
  incrementQuantity,
  decrementQuantity,
  setVariant,
};


export default connect(mapStateToProps, mapDispatchToProps) (PopCart);

