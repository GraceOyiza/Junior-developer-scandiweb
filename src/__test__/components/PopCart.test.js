import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PopCart from '../../components/PopCart';
import { Provider } from 'react-redux';
import store from '../../app/store';
import { BrowserRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
    wrapper = render(
      <BrowserRouter>
        <Provider store={store}>
          <PopCart />
        </Provider>
      </BrowserRouter>,
    );
});

describe('',  () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
     
})