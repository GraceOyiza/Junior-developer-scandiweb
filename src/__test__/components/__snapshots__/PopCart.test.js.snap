// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
LoadedCheerio {
  "0": <div>
    <div
      class="sc-jrcTuL jgXOcc"
    />
    <div
      class="sc-jSUZER kWqyiV"
    >
      <div />
      <h3>
        My Bag 0
      </h3>
      <div>
        <div>
          <p>
            Total: $0.00
          </p>
          <a
            href="/cart"
          >
            <button
              class="sc-pyfCe jvCCuv"
            >
              VIEW BAG
            </button>
          </a>
          <button
            class="sc-ftTHYK bZWAXr"
          >
            CHECK OUT
          </button>
        </div>
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
